src/ui/map.js

```diff
@@ -171,8 +171,9 @@ const defaultOptions = {
  * Tilesets hosted with Mapbox can be style-optimized if you append `?optimize=true` to the end of your style URL, like `mapbox://styles/mapbox/streets-v9?optimize=true`.
  * Learn more about style-optimized vector tiles in our [API documentation](https://www.mapbox.com/api-documentation/maps/#retrieve-tiles).
  *
- * @param {boolean} [options.hash=false] If `true`, the map's position (zoom, center latitude, center longitude, bearing, and pitch) will be synced with the hash fragment of the page's URL.
+ * @param {(boolean|string)} [options.hash=false] If `true`, the map's position (zoom, center latitude, center longitude, bearing, and pitch) will be synced with the hash fragment of the page's URL.
  *   For example, `http://path/to/my/page.html#2.59/39.26/53.07/-24.1/60`.
+ *   If a string is provided, it will be used as the name for a parameter-styled hash, e.g. `http://path/to/my/page.html#map=2.59/39.26/53.07/-24.1/60&foo=bar`.
```

Comment:
Nit, but I think it's probably best to leave the `&foo=bar` off the example just to be clear that something like that isn't added by using a string in this property. You could expand the comment though to include an explanation like "This allows the developer to use arbitrary hashes in addition to the map hash`.

cc @chloekraw on thoughts/wording for this?

Reply:
Agree with @ryanhamley that as-is, there is some potential for confusion. I would suggest explicitly flagging this as optional and adding a clause at the end of what you've written to explain what `foo` and `bar` refer to. My proposal:

> An additional string may optionally be provided to indicate a parameter-styled hash, e.g. `http://path/to/my/page.html#map=2.59/39.26/53.07/-24.1/60&foo=bar`, where `foo` is a custom parameter and `bar` is an arbitrary hash distinct from the map hash.

Also, could we fix up the linebreaking throughout the PR? We generally keep comments to the 80 column mark.